scripts/filter_multimappers.py

```diff
@@ -133,13 +134,20 @@ def main(sam_file: Path) -> None:
         sam_file: Path to the input SAM file.
     """
     with pysam.AlignmentFile(sam_file, "r") as samfile:
+      
+        try:
+            first_alignment = next(samfile)
+            
+        except StopIteration:
+            sys.stdout.write(str(samfile.header))
+            return
 
         sys.stdout.write(str(samfile.header))
 
         current_query = None
         current_alignments: list[pysam.AlignedSegment] = []
 
-        for alignment in samfile:
+        for alignment in chain([first_alignment],samfile):
 
             if alignment.is_secondary or alignment.is_supplementary:
                 continue
```

Comment:
Can you remind me again why we are doing this? Doesn't this mean that we are _always_ just keeping only a single alignment and never even call `find_best_alignment()`?

That is, if the "secondary alignment" flag is actually set correctly (`0` for the first alignment of a given query name, `1` for every other alignment)...

Not sure what "supplementary alignments" are - this you can probably leave.

Reply:
Regarding the `is_secondary` statement, you are right, already remove it. As for the supplementary alignments, this FLAG is for alignments that align "by parts"; one of the aligned segments will be considered the primary alignment and the rest will be considered supplementary.  [Here](https://www.biostars.org/p/149883/) it is quite well explained.